hoomd/logging.py

```diff
@@ -89,64 +267,268 @@ def update_cls(self, cls):
         """Allow updating the class/namespace of the object.
 
         Since the namespace is determined by the passed class's module and class
-        name, if inheritanting `hoomd.logging.LoggerQuantity`, the class needs
+        name, if inheritanting `hoomd.logging._LoggerQuantity`, the class needs
         to be updated to the subclass.
 
         Args:
             cls (``class object``): The class to update the namespace with.
         """
-        self.namespace = generate_namespace(cls)
+        self.namespace = self._generate_namespace(cls)
         return self
 
+    @staticmethod
+    def _generate_namespace(cls):
+        """Infite iterator of namespaces for a given class.
+
+        If namespace is taken add a number and increment until unique.
+        """
+        ns = tuple(cls.__module__.split('.') + [cls.__name__])
+        if ns[0] == 'hoomd':
+            return ns[1:]
+        else:
+            return ns
+
+
+class _LoggerEntry:
```

Comment:
I'm not completely sure, but this class seems a little overengineered. The two factory methods are each called exactly once, and I'm not sure of the purpose of the `__eq__` and `__call__` methods. Maybe it's because I'm not looking at how the representation gets parsed by the specific backends, so I'm just trying to provide a balance against introducing unnecessary complexity and levels of abstraction where it isn't necessarily merited.

Reply:
The `__call__` method deals with methods and properties correctly, and appending the flag correctly. I made the separation because it gets really noisy in the `Logger`. I also for see that we may want to validate that the user created loggable quantites are the flag they specify.